src/testing/TestFields.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { FC } from 'react';
+import { Box, Text, Label, Input, Checkbox, Textarea, Radio } from 'theme-ui';
+import { FormikContextType, FormikValues } from 'formik';
+import { Message } from 'retranslate';
+import * as yup from 'yup';
+
+import { TestType, ObjectSchema, FieldSchema } from '../api';
+
+interface TestFieldsProps {
+  form: FormikContextType<FormikValues>;
+  testType: TestType;
+}
+
+export const TestFields: FC<TestFieldsProps> = ({ form, testType }) => {
+  const fieldError = (key: keyof TestType['resultsSchema']['properties']) =>
+    form.touched[key] && form.errors[key] ? <Text>{form.errors[key]}</Text> : null;
+
+  return (
+    <>
+      {Object.entries(testType.resultsSchema.properties).map(([key, value]) => {
+        if (value.oneOf) {
+          return (
+            <Box key={key}>
+              {value.title} {testType.resultsSchema.required?.includes(key) && '*'}
+              {value.oneOf.map((option) => (
+                <Label key={`${option.const}`}>
+                  <Radio
+                    checked={form.getFieldProps(key).value === option.const}
+                    value={typeof option.const === 'number' ? option.const : `${option.const}`}
+                    onChange={({ target }) =>
+                      form.setFieldValue(
+                        key,
+                        typeof option.const === 'boolean' ? target.value === 'true' : target.value
```

Comment:
Why do you need this, rather than just `option.const` ?

Reply:
https://github.com/cov-clear/web/pull/62#discussion_r421372523